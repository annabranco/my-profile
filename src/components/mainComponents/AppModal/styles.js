/* eslint-disable no-unused-vars */

import styled, { css } from 'styled-components';
import { rgba } from 'polished';
import {
  colorBlack,
  colorBlueDark,
  colorWhite,
  fontTitleAlt
} from '../../../styles/theme';

export const CloseButton = styled.p`
  z-index: 100;
  position: fixed;
  top: 0;
  right: 0;
  z-index: 100;
  color: red;
  color: ${({ colors }) => colors.terciary};
  background: ${({ colors }) => colors && colors.primary};
  cursor: pointer;
  border-radius: 5px;
  padding: 5px 10px;

  &:hover {
    box-shadow: 0 0 2px 2px white;
  }

  ${({ text }) =>
    text &&
    css`
      &::before {
        content: 'x';
        text-transform: uppercase;
        font-weight: 700;
        font-family: ${fontTitleAlt};
        font-size: 1rem;
      }
    `};

  @media all and (min-width: 768px) {
    ${({ colors, text }) =>
      text &&
      css`
        top: 20px;
        right: 50px;
        border: ${`2px solid ${colors.terciary}`};
        background: none;

        &::before {
          content: '${text}';
          text-transform: uppercase;
          font-weight: 700;
        }
      `};
  }
`;
CloseButton.displayName = 'CloseButton';

export const ModalOverlay = styled.div`
  position: fixed;
  top: 0;
  left: 0;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  height: ${`${window.innerHeight}px`};
  overflow: hidden;
  width: 100vw;
  z-index: 100;
  background-image: linear-gradient(
    to bottom,
    ${rgba(colorBlueDark, 0.9)},
    ${rgba(colorBlueDark, 0.9)},
    ${rgba(colorBlack, 1)}
  );
  padding: 0;
`;
ModalOverlay.displayName = 'ModalOverlay';
