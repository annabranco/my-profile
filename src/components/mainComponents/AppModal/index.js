/* eslint-disable import/no-extraneous-dependencies */
import React from 'react';
import { node, func } from 'prop-types';
import { ModalOverlay } from './styles';

const AppModal = ({ children, closeAction }) => {
  return <ModalOverlay onClick={closeAction}>{children}</ModalOverlay>;
};

AppModal.propTypes = {
  children: node,
  closeAction: func
};

AppModal.defaultProps = {
  children: undefined,
  closeAction: () => null
};

export default AppModal;
