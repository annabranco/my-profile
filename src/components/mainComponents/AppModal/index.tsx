/* eslint-disable import/no-extraneous-dependencies */
import { ReactElement, ReactNode } from 'react';

import * as React from 'react';
import { func, node } from 'prop-types';
import { ModalOverlay } from './styles';

interface Props {
  children: ReactNode | ReactNode[];
  closeAction?: (event: React.MouseEvent) => void;
}

const AppModal = ({ children, closeAction }: Props): ReactElement => {
  return <ModalOverlay onClick={closeAction}>{children}</ModalOverlay>;
};

AppModal.propTypes = {
  children: node,
  closeAction: func
};

AppModal.defaultProps = {
  children: undefined,
  closeAction: () => null
};

export default AppModal;
