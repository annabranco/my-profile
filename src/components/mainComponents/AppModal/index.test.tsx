import { shallow, ShallowWrapper } from 'enzyme';
import 'jest-styled-components';
import AppModal from '.';

// TODO [10-May -21]: Implement defined testing suite (Anna Branco)

const MockChild = () => <p>Mock child</p>;

describe('< AppModal >', () => {
  let wrapper: ShallowWrapper;

  beforeAll(() => {
    wrapper = shallow(
      <AppModal closeAction={() => jest.fn()}>
        <MockChild />
      </AppModal>
    );
  });

  it('should mount', () => {
    expect(wrapper).toBeDefined();
  });

  it('matches the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  xit('should render an overlay on the top of the screen', () => {
  });

  xit('should correctly render any children received by props', () => {
  });

  xit('should trigger a close action when clicked anywhere', () => {
  });
});
