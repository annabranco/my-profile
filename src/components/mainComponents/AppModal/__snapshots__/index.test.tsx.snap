// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`< AppModal > matches the snapshot 1`] = `
<ModalOverlay
  onClick={[Function]}
>
  <MockChild />
</ModalOverlay>
`;
